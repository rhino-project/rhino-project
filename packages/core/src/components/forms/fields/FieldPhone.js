import PropTypes from 'prop-types';
import classnames from 'classnames';
import { useController } from 'react-hook-form';
import { useMemo } from 'react';

import { useFieldInheritedProps } from '../../../hooks/form';
import PhoneInput from 'react-phone-input-2';
import { useGlobalComponent } from '../../../hooks/overrides';

export const FieldPhoneBase = (props) => {
  const { path } = props;
  const {
    field: fieldProps,
    fieldState: { error }
  } = useController({
    name: path
  });

  const containerClass = useMemo(
    () => classnames({ 'is-invalid': error }),
    [error]
  );

  const { extractedProps, inheritedProps } = useFieldInheritedProps(props);

  return (
    <PhoneInput
      containerClass={containerClass}
      {...extractedProps}
      {...fieldProps}
      country="us"
      preferredCountries={['us', 'ca']}
      regions={['north-america', 'europe']}
      {...inheritedProps}
    />
  );
};

FieldPhoneBase.propTypes = {
  path: PropTypes.string.isRequired
};

export const FieldPhone = (props) =>
  useGlobalComponent('FieldPhone', FieldPhoneBase, props);
