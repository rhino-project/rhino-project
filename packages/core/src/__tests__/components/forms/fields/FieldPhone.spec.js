import FieldPhone from '../../../../components/forms/fields/FieldPhone';
import { sharedFieldTests } from './sharedFieldTests';

describe('FieldPhone', () => {
  sharedFieldTests(FieldPhone);
});
