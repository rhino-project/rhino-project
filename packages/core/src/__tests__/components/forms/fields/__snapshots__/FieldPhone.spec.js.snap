// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FieldPhone > render with disabled 1`] = `
<DocumentFragment>
  <div
    class=" react-tel-input "
  >
    <div
      class="special-label"
    >
      Phone
    </div>
    <input
      class="form-control "
      disabled=""
      placeholder="1 (702) 123-4567"
      type="tel"
      value="+1"
    />
    <div
      class="flag-dropdown "
    >
      <div
        aria-haspopup="listbox"
        class="selected-flag"
        role="button"
        tabindex="0"
        title="United States: + 1"
      >
        <div
          class="flag us"
        >
          <div
            class="arrow"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FieldPhone > render with place holder 1`] = `
<DocumentFragment>
  <div
    class=" react-tel-input "
  >
    <div
      class="special-label"
    >
      Phone
    </div>
    <input
      class="form-control "
      placeholder="placeholder"
      type="tel"
      value="+1"
    />
    <div
      class="flag-dropdown "
    >
      <div
        aria-haspopup="listbox"
        class="selected-flag"
        role="button"
        tabindex="0"
        title="United States: + 1"
      >
        <div
          class="flag us"
        >
          <div
            class="arrow"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FieldPhone > renders inside of model context 1`] = `
<DocumentFragment>
  <div
    class=" react-tel-input "
  >
    <div
      class="special-label"
    >
      Phone
    </div>
    <input
      class="form-control "
      placeholder="1 (702) 123-4567"
      type="tel"
      value="+1"
    />
    <div
      class="flag-dropdown "
    >
      <div
        aria-haspopup="listbox"
        class="selected-flag"
        role="button"
        tabindex="0"
        title="United States: + 1"
      >
        <div
          class="flag us"
        >
          <div
            class="arrow"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FieldPhone > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
